Test branch to main

mmk150 opened the pull request:
This branch should have slash commands implemented correctly. A quick double check that nothing has gone horribly wrong with the NixOs bits would be appreciated.

mmk150 committed:
Testing implementation of slash commands

mmk150 committed:
Hotfix

mmk150 committed:
Tidying up test branch with some updates

mmk150 committed:
Tidying up test branch-typo fix.

LunNova commented on src/thoth/commandio.py:
```diff
@@ -1,147 +1,221 @@
-from thoth import databaseio
+import databaseio
```
change this back
```suggestion
from thoth import databaseio
```

LunNova replied:
same for timecalcs and timerz

LunNova commented:
After fixing the imports got this error when tested something:

```
2023-02-07 13:57:47 ERROR    discord.ext.tasks Unhandled exception in internal background task 'check_timers'.
Traceback (most recent call last):
  File "/nix/store/kdnyk68ykwx24w46xnkwjjra85g30k2j-python3-3.10.7-env/lib/python3.10/site-packages/discord/ext/tasks/__init__.py", line 239, in _loop
    await self.coro(*args, **kwargs)
  File "/nix/store/kdnyk68ykwx24w46xnkwjjra85g30k2j-python3-3.10.7-env/lib/python3.10/site-packages/thoth/main.py", line 382, in check_timers
    await send_late_reminder(timer_obj)
  File "/nix/store/kdnyk68ykwx24w46xnkwjjra85g30k2j-python3-3.10.7-env/lib/python3.10/site-packages/thoth/main.py", line 491, in send_late_reminder
    await channeler.send(full_message)
AttributeError: 'NoneType' object has no attribute 'send'
```

LunNova committed:
fix imports and try to make reminders more robust

LunNova committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'flake-compat':
    'github:edolstra/flake-compat/b4a34015c698c7793d592d66adbab377907a2be8' (2022-04-19)
  → 'github:edolstra/flake-compat/35bb57c0c8d8b62bbfd284272c928ceb64ddbde9' (2023-01-17)
• Updated input 'flake-utils':
    'github:numtide/flake-utils/c0e246b9b83f637f4681389ecabcb2681b4f3af0' (2022-08-07)
  → 'github:numtide/flake-utils/93a2b84fc4b70d9e089d029deacc3583435c2ed6' (2023-03-15)
• Updated input 'nixpkgs':
    'github:nixos/nixpkgs/95aeaf83c247b8f5aa561684317ecd860476fcd6' (2022-10-22)
  → 'github:nixos/nixpkgs/19cf008bb18e47b6e3b4e16e32a9a4bdd4b45f7e' (2023-03-21)
